pkg/proc/bininfo.go

```diff
@@ -455,23 +514,33 @@ func (bi *BinaryInfo) LoadBinaryInfoElf(path string, wg *sync.WaitGroup) error {
 	if elfFile.Machine != elf.EM_X86_64 {
 		return UnsupportedLinuxArchErr
 	}
+	debugInfo := elfFile
 	bi.dwarf, err = elfFile.DWARF()
 	if err != nil {
-		return err
+		var sepFile *os.File
+		sepFile, debugInfo, err = bi.LoadSeparateDebugInfo(elfFile)
+		if err != nil {
+			return err
```

Comment:
Hm... I'm a bit lost here, openSeparateDebugInfo already returns a variety of errors here depending on the actual issue dealing with the file.

Reply:
Suppose you have an executable with embedded DWARF (no build-id note). Also suppose that calling elfFile.DWARF() returns an error for this file. What you want to return to the user is the error returned from elfFile.DWARF() not the "can't find build-id note" error, the former is informative, the latter is misleading (we only looked for build-id because of the error returned by elfFile.DWARF()).

Reply:
Now I get it, thanks a lot for the explanation. I've just fixed this one too.